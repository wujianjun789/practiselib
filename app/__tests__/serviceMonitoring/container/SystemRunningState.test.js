/**
 * Created by a on 2017/10/9.
 */
import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer';

import SystemRunningState from '../../../src/serviceMonitoring/container/SystemRunningState';

const monitorList = [
    {id:"cpu", name:"CPU", detail:"INTEL(R) XEON(R) CPU E5-1620 V3"},
    {id:"memory", name:"内存", detail:"16.0GB"},
    {id:"disk", name:"磁盘", detail:"500G"},
    {id:"network", name:"网络", detail:"INTEL(R) ETHER CONNECTION L217-LM"}
]

test('renders Scene', ()=>{
    const component = renderer.create(
        <SystemRunningState />
    ).toJSON();

    expect(component).toMatchSnapshot();
});

test('renders Scene data', ()=>{
    const component = shallow(<SystemRunningState />);
    component.setState({ monitorList:monitorList});
    expect(component.find('.monitor').length).toEqual(monitorList.length);
})