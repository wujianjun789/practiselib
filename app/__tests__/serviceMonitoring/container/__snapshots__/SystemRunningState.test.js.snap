exports[`test renders Scene 1`] = `
<div
  className="content ">
  <div
    className="col-sm-6 monitor">
    <div
      className="heading">
      <span
        className="name">
        CPU
      </span>
      INTEL(R) XEON(R) CPU E5-1620 V3
      <span
        className="right">
        100%
      </span>
    </div>
    <div
      className="chart cpu"
      id="cpu" />
  </div>
  <div
    className="col-sm-6 monitor">
    <div
      className="heading">
      <span
        className="name">
        内存
      </span>
      16.0GB
      <span
        className="right">
        100%
      </span>
    </div>
    <div
      className="chart memory"
      id="memory" />
  </div>
  <div
    className="col-sm-6 monitor">
    <div
      className="heading">
      <span
        className="name">
        磁盘
      </span>
      500G
      <span
        className="right">
        100%
      </span>
    </div>
    <div
      className="chart disk"
      id="disk" />
  </div>
  <div
    className="col-sm-6 monitor">
    <div
      className="heading">
      <span
        className="name">
        网络
      </span>
      INTEL(R) ETHER CONNECTION L217-LM
      <span
        className="right">
        100%
      </span>
    </div>
    <div
      className="chart network"
      id="network" />
  </div>
</div>
`;
